### PR TITLE
doc(examples): Update examples with groups in slices

### PR DESCRIPTION
**Pull request by xrutayisire**
## Types of changes

- [x] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

- Completes [DT-2082](https://linear.app/prismic/issue/DT-2082/aadevuser-i-can-see-the-examples-of-prismicioreact-updated-with-a)
- Updated examples to use `SharedSlice`
- Updated examples to use a Group instead of the `items` property for a `ShareSlice`

## Checklist:

- [ ] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

### CODE REVIEW

File: examples/with-typescript/index.tsx [hunk 1]
```diff
@@ -11,34 +11,38 @@ type PageDocument = prismic.PrismicDocumentWithUID<{
...
-					buttonText: prismic.KeyTextField;
-					buttonLink: prismic.LinkField;
-				}
+				prismic.SharedSliceVariation<
+					"default",
+					// Fields
+					{
+						heading: prismic.KeyTextField;
+						body: prismic.RichTextField;
+						items: prismic.GroupField<{
```
Comment (angeloashmore) [inline examples/with-typescript/index.tsx:25]: 💡 Suggestion: replace `items` with `buttons` to describe its contents. Furthermore, we can remove the `button` prefix on the group's fields:

```ts
buttons: prismic.GroupField<{
  label: prismic.KeyTextField;
  link: prismic.KeyTextField;
}>
```

File: examples/with-typescript/index.tsx [hunk 2]
```diff
@@ -11,34 +11,38 @@ type PageDocument = prismic.PrismicDocumentWithUID<{
...
-				{
-					buttonText: prismic.KeyTextField;
-					buttonLink: prismic.LinkField;
-				}
+				prismic.SharedSliceVariation<
+					"default",
+					// Fields
+					{
+						text: prismic.RichTextField;
+						items: prismic.GroupField<{
```
Comment (angeloashmore) [inline examples/with-typescript/index.tsx:40]: 💡 Similar to above, we could replace `items` with `buttons` as a better example.

**Review by angeloashmore — APPROVED** [review]
Looks great! 🙂 I left two suggestions below. It's good to merge after you consider them. 👍

File: examples/with-typescript/index.tsx (hunk 1 above)
Comment (xrutayisire) [reply examples/with-typescript/index.tsx]: Good idea! Done 🙂

File: examples/with-typescript/index.tsx (hunk 2 above)
Comment (xrutayisire) [reply examples/with-typescript/index.tsx]: Good idea! Done 🙂

**Pull request merged**